test/WebSites/TagHelpersWebSite/Views/Catalog_CacheTagHelper/Splash.cshtml

```diff
@@ -0,0 +1,8 @@
+﻿<h2>Category: @ViewBag.Category</h2>
+<h2>Region: @ViewBag.Locale</h2>
+<cache asp-vary-by-header="Locale" asp-vary-by-query="categoryid">
+    <h2>Cached content</h2>
+    @await Component.InvokeAsync("Splash")
+    @await Html.PartialAsync("_SplashPartial")
+    <div>CorrelationId in Splash: @ViewBag.CorrelationId</div>
```

Comment:
Spoke to @DamianEdwards offline. We decided on going with the first approach. I'll follow up with a separate PR to reduce noise in this one.